tests/nosetests/pyanaconda_tests/module_source_repo_files_test.py

```diff
@@ -0,0 +1,125 @@
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+import unittest
+from unittest.mock import patch
+
+from pyanaconda.modules.common.errors.payload import SourceSetupError
+from pyanaconda.modules.payloads.constants import SourceType
+from pyanaconda.modules.payloads.source.repo_files.repo_files import RepoFilesSourceModule
+from pyanaconda.modules.payloads.source.repo_files.repo_files_interface import \
+    RepoFilesSourceInterface
+from pyanaconda.modules.payloads.source.repo_files.initialization import \
+    SetUpRepoFilesSourceTask, TearDownRepoFilesSourceTask
+
+from tests.nosetests.pyanaconda_tests import PropertiesChangedCallback
+
+
+class RepoFilesSourceInterfaceTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.install_media_source_module = RepoFilesSourceModule()
+        self.install_media_source_interface = RepoFilesSourceInterface(
+            self.install_media_source_module
+        )
+
+        self.callback = PropertiesChangedCallback()
+        self.install_media_source_interface.PropertiesChanged.connect(self.callback)
+
+    def type_test(self):
+        """Test Repo files source has a correct type specified."""
+        self.assertEqual(SourceType.REPO_FILES.value, self.install_media_source_interface.Type)
+
+
+class RepoFilesSourceTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.install_media_source_module = RepoFilesSourceModule()
+
+    def type_test(self):
+        """Test Repo files source module has a correct type."""
+        self.assertEqual(SourceType.REPO_FILES, self.install_media_source_module.type)
+
+    def set_up_with_tasks_test(self):
+        """Test Repo files Source set up call."""
+        task_classes = [
+            SetUpRepoFilesSourceTask
+        ]
+
+        # task will not be public so it won't be published
+        tasks = self.install_media_source_module.set_up_with_tasks()
+
+        # Check the number of the tasks
+        task_number = len(task_classes)
+        self.assertEqual(task_number, len(tasks))
+
+        for i in range(task_number):
+            self.assertIsInstance(tasks[i], task_classes[i])
```

Comment:
You can replace this by the `check_task_creation_list()` function.

Reply:
I replaced this with a shorter version.